src/adapter/src/coord/sequencer.rs

```diff
@@ -1401,6 +1401,16 @@ impl Coordinator {
         plan: CreateViewPlan,
         depends_on: Vec<GlobalId>,
     ) -> Result<ExecuteResponse, AdapterError> {
+        if plan.ambiguous_columns
+            && depends_on
+                .iter()
+                .any(|id| id.is_system() && self.catalog.get_entry(id).is_relation())
+        {
+            return Err(AdapterError::Unsupported(
+                "ambiguous column references, like NATURAL JOIN or SELECT *, in a view with system objects",
```

Comment:
I think it'd be worth promoting this to a full blown error variant with a hint. Also they're not _technically_ ambiguous column references _yet_, right? It's more that they have the potential to become ambiguous? So what about this?

```
ERROR: cannot use wildcard expansions or NATURAL JOINs in a view that depends on system objects
DETAIL: This is a limitation in Materialize that will be lifted in a future release. See https://github.com/MaterializeInc/materialize/issues/WHATEVER for details.
HINT: Rewrite the view to refer to all columns by name. Expand all wildcards and convert all NATURAL JOINs to USING joins.
```

Reply:
Done